packages/@vuepress/markdown/__tests__/plugins/assetsPlugin.spec.ts

```diff
@@ -189,6 +190,41 @@ describe('@vuepress/markdown > plugins > assetsPlugin', () => {
       '<img src=".../汉字.png">',
       '<img src=".../100%.png">',
       '<img alt="attrs" src=".../attrs.png" width="100px">',
+
+      /* srcset */
+
+      // relative paths
+      '<img srcset="./foo.png 1x, ../sub/foo.png 2x, ./foo/bar.png 1024w ,../sub/foo/bar.png 2048w, ../baz.png 4096w,../../out.png">',
+      '<img srcset="./汉字.png 1x, ./100%.png">',
+      '<img alt="attrs" srcset="./attrs.png" width="100px">',
+      // aliases
+      '<img srcset="@alias/foo.png 1x,@alias/汉字.png 2x, @alias/100%.png 3x">',
+      '<img alt="attrs" srcset="@alias/attrs.png 1024w" width="100px">',
+      // webpack legacy aliases
+      '<img srcset="~@alias/foo.png 1x,~@alias/汉字.png 2x, ~@alias/100%.png 3x">',
+      '<img alt="attrs" srcset="~@alias/attrs.png 1024w" width="100px">',
+      // keep as is
+      '<img srcset="/absolute.png 1x, no-prefix.png 2x, http://foobar.com/icon.png">',
+      '<img srcset="">',
+      '<img alt="attrs" srcset="attrs.png 1x, default.png" width="100px">',
+      // invalid paths
+      '<img srcset=".../invalid.png 1x, .../汉字.png 2x, .../100%.png 3x">',
+      '<img alt="attrs" srcset=".../attrs.png 1x, .../default.png" width="100px">',
+      // invalid srcset
+      '<img srcset="../invalid.png, ../汉字.png, .../100%.png 3x">',
+
+      /** both */
+      // relative paths
+      '<img srcset="./foo.png 1x, ../sub/foo.png 2x, ./foo/bar.png 1024w ,../sub/foo/bar.png 2048w, ../baz.png 4096w,../../out.png 3x" src="./default.png">',
+      '<img src="./100%.png"  srcset="./汉字.png 1x" >',
```

Comment:
Extra whitespace before `srcset`. Check the format again

Reply:
That's expected in my design either, users may not  have standard input.

Reply:
Then you need to add other cases for that purpose.

Reply:
For example, `non-formatted` to mix those cases.

Reply:
Emm, well, I will add those. Just think that it was not really necessary to add a lot of test cases which are acually unneeded.